### ISSUE TITLE
Additional permission popup appears for one time on enabling 'ga4reporting' feature flag.

### ISSUE THREAD

**Issue by mohitwp** [google/site-kit-wp#6806]
## Bug Description

Now Additional permission pop up not appearing when we enabled `'gteSupport`' feature flag. But I've noticed that when we enable `ga4reporting` feature flag then for one time Additional permission popup appears. After that if I refresh the main dashboard or disabled and again enabled the 'ga4Reporting feature flag then pop up gets hide.

## Steps to reproduce

1. Set up Analytics.
2. Enabled 'gteSupport' feature flag.
3. Now enabled 'ga4Reporting' feature flag.
4. See 'Additional permission popup' appears on main dashboard.
5. Now refresh the page.
6. Popup gets hide.

## Screenshots

https://user-images.githubusercontent.com/94359491/229162676-7948c1cf-9a5f-4c21-871e-31f160afc3e7.mp4

## Additional Context

 - PHP Version: 
 - Browser: [e.g. chrome, safari]
 - Plugin Version: [e.g. 22]

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by eclarke1** [comment]
@aaemnnosttv can we please get the AC moving on this one ASAP please cc @bethanylang

**Comment by aaemnnosttv** [comment]
@aaemnnosttv Yes, I spoke to @felixarntz about it yesterday and I think we've agreed on a way forward here.

**Comment by felixarntz** [comment]
@aaemnnosttv Why do we need a new `googlesitekit_auth_scopes` filter in the `Analytics_4` module class? Couldn't the same conditional logic be handled directly in the `Analytics_4::get_scopes()` method?

**Comment by aaemnnosttv** [comment]
This is essentially to avoid this scope from breaking the request validation in `validate_base_scopes` which also uses `get_scopes`. I think if we did it in `get_scopes` we'd need to operate on the module owner's scopes for it to work in a shared context which would not be the context we'd want when conditionally determining whether to require the scope for the current user or not.

So this is a way to conditionally require the scope while decoupling it from being a base scope of the module, while still kind of treating it as one, even though it's only required for a few specific requests.

**Comment by felixarntz** [comment]
@aaemnnosttv Makes sense, thanks! IB ✅

**Comment by mohitwp** [comment]
QA Update ✅ 

- Tested on dev environment.
- Verified additional permission popup not appears when both 'gteSupport' and 'ga4Reporting' feature flag is enabled.
- Performed a regression test of the connection process for Site Kit and Analytics, with the ga4Reporting and gteSupport feature flags in different states, and connecting Analytics both as part of the initial setup flow and separately, and as a primary and secondary admin.

  - When 'ga4Reporting' feature flag is enabled.
  - When only 'gteSupport' feature flag is enabled.
  - When both 'ga4Reporting' and  'gteSupport' feature flag is enabled.
  - When both 'ga4Reporting' and  'gteSupport' feature flag are not enabled.
  
- Performed a regression test that makes use of the container-lookup and container-destinations endpoints. By, following the steps under "Check GA4 measurement ID mapping from Google Tag periodically" in the [GTE Bug Bash instructions](https://app.asana.com/0/1203993135029815/1204174775213831) to display the notification.
 
-Smoke tested the view-only Dashboard with Analytics shared and the ga4Reporting and gteSupport feature flags in different states to make sure there are no unexpected requests for permissions or permission-related console errors.

   - When 'ga4Reporting' feature flag is enabled.
  - When only 'gteSupport' feature flag is enabled.
  - When both 'ga4Reporting' and  'gteSupport' feature flag is enabled.
  - When both 'ga4Reporting' and  'gteSupport' feature flag are not enabled.

Verified Acceptance Criteria.

<details>



https://github.com/google/site-kit-wp/assets/94359491/e128a5db-b069-4522-bc03-00f2b8ed2317


https://github.com/google/site-kit-wp/assets/94359491/a1f03fbc-4298-44ce-a96d-4a299c37b650


![image](https://github.com/google/site-kit-wp/assets/94359491/9caa16bc-8828-4805-a216-afdb5f2c8ebb)

</details>

**Comment by aaemnnosttv** [comment]
LGTM 👍
